rest-api-guidelines/functionality/message/error-reporting.md

```diff
@@ -122,5 +124,8 @@ A Problem Detail response **MUST NOT** contain a program stack trace or server l
 
 ## Working with Problem Detail
 
-There are a whole plethora of libraries working with Problem Detail, for example, see [Zalando / Problem](https://github.com/zalando/problem) \(Java\).
+An API description **SHOULD** list all the error codes with which the API responds. The error responses **SHOULD** describre the error object model schema. It is **RECOMMENDED** to include examples of a possible error response. The error description and/or error example **MAY** list all the types of errors returned for a given error code.
```

Comment:
This paragraph is the first time you introduce the term `error code` in this document. Initially, I was afraid that you meant the HTTP Status codes. I think it is essential to rephrase it and say "problem detail types" or "problem types" to prevent any confusion.

As for as what HTTP layer - errors to document, I generally tend to recommend not to document HTTP-defined errors (e.g., not found)  unless they are specific to the API.

Reply:
so you say you recommend not to list whether an API returns e.g. 409? Why is so? I'd say it may be pretty important to know how it responds for different issues with requests. What am I missing?

As for the problem detail type - agree.

Reply:
An API, that is compliant to HTTP protocol CAN return ANY standard HTTP status code. 

I do not want to document dozens of protocol-given codes/response if there are 20 endpoints in an API description on every endpoint! Same goes to HTTP headers. You do not want to describe standard HTTP headers (`Content-Length` anyone?).

I only recommend documenting non-standard, differently used or otherwise important responses/codes. Everything else needs to be HTTP protocol driven. Does it clarify?

Reply:
In other words, I suggest to  rephrase the following:

> list all the error codes with which the API responds